browser/extensions/api/brave_rewards_api.cc

```diff
@@ -600,47 +613,65 @@ BraveRewardsCreateRewardsWalletFunction::
 
 ExtensionFunction::ResponseAction
 BraveRewardsCreateRewardsWalletFunction::Run() {
+  auto params = brave_rewards::CreateRewardsWallet::Params::Create(args());
+  EXTENSION_FUNCTION_VALIDATE(params);
+
   auto* profile = Profile::FromBrowserContext(browser_context());
   auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
   if (!rewards_service) {
     return RespondNow(Error("RewardsService not available"));
   }
 
-  rewards_service->CreateRewardsWallet(base::BindOnce(
-      &BraveRewardsCreateRewardsWalletFunction::CreateRewardsWalletCallback,
-      this));
+  rewards_service->CreateRewardsWallet(
+      params->country,
+      base::BindOnce(
+          &BraveRewardsCreateRewardsWalletFunction::CreateRewardsWalletCallback,
+          this));
 
   return RespondLater();
 }
 
 void BraveRewardsCreateRewardsWalletFunction::CreateRewardsWalletCallback(
-    ledger::mojom::Result result) {
-  Respond(OneArgument(base::Value(static_cast<int>(result))));
+    ledger::mojom::CreateRewardsWalletResult result) {
+  Respond(OneArgument(base::Value(StringifyResult(result))));
 }
 
-BraveRewardsGetRewardsWalletFunction::~BraveRewardsGetRewardsWalletFunction() =
-    default;
+BraveRewardsGetAvailableCountriesFunction::
+    ~BraveRewardsGetAvailableCountriesFunction() = default;
 
-ExtensionFunction::ResponseAction BraveRewardsGetRewardsWalletFunction::Run() {
-  Profile* profile = Profile::FromBrowserContext(browser_context());
+ExtensionFunction::ResponseAction
+BraveRewardsGetAvailableCountriesFunction::Run() {
+  auto* profile = Profile::FromBrowserContext(browser_context());
   auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
+
   if (!rewards_service) {
-    return RespondNow(Error("RewardsService not available"));
+    return RespondNow(Error("Rewards service is not initialized"));
   }
 
-  rewards_service->GetRewardsWallet(base::BindOnce(
-      &BraveRewardsGetRewardsWalletFunction::GetRewardsWalletCallback, this));
+  rewards_service->GetAvailableCountries(base::BindOnce(
+      &BraveRewardsGetAvailableCountriesFunction::GetAvailableCountriesCallback,
+      this));
+
   return RespondLater();
 }
 
-void BraveRewardsGetRewardsWalletFunction::GetRewardsWalletCallback(
-    ledger::mojom::RewardsWalletPtr rewards_wallet) {
-  if (!rewards_wallet) {
-    return Respond(NoArguments());
+void BraveRewardsGetAvailableCountriesFunction::GetAvailableCountriesCallback(
+    std::vector<std::string> countries) {
+  base::Value::List country_list;
+  for (const auto& country : countries) {
```

Comment:
This should be reference-to-non-`const` (you're moving `country`).

Reply:
Fixed